addons/calendar/models/calendar_event.py

```diff
@@ -346,14 +345,44 @@ def _compute_recurrence(self):
             else:
                 event.update(false_values)
 
+    @api.depends('attendee_ids')
+    def _compute_partners(self):
+        for event in self:
+            partner_ids = event.attendee_ids.mapped('partner_id')
+            event.partner_ids = [(6, 0, partner_ids.ids)]
```

Comment:
So complicated...  Simply assign the records.  And `mapped` is built-in when you read relational fields.
```python
event.partner_ids = event.attendee_ids.partner_id
```